utils/s3/client.cc

```diff
@@ -112,8 +129,13 @@ shared_ptr<client> client::make(std::string endpoint, endpoint_config_ptr cfg, s
 }
 
 future<> client::authorize(http::request& req) {
-    if (!_credentials || std::chrono::system_clock::now() >= _credentials.expires_at) {
-        _credentials = co_await _creds_provider_chain.get_aws_credentials();
+    if (!_credentials) {
+        co_await _creds_sem.wait();
```

Comment:
nit: better to use RAII get_units() helper. If the below `get_aws_credentials()` throws, the semaphore remains locked forever :skull_and_crossbones:

Reply:
@xemul - why did you marked it with `nit`?

Reply:
Because I first decided that it was a nit, and while typing it realized that it was also exception unsafe :D